osu.Game/Modes/Ruleset.cs

```diff
@@ -51,5 +58,28 @@ public static Ruleset GetRuleset(PlayMode mode)
 
             return Activator.CreateInstance(type) as Ruleset;
         }
+        
+        [DebuggerNonUserCode]
+        public static void LoadRulesetsFrom(string directory)
+        {
+            foreach (string subDirectory in Directory.EnumerateDirectories(directory))
+                loadRulesets(subDirectory);
+
+            foreach (string file in Directory.EnumerateFiles(directory))
+            {
+                if (!file.EndsWith(".dll"))
+                    continue;
+                try
+                {
+                    var rulesets = Assembly.LoadFile(file).GetTypes().Where((Type t) => t.IsSubclassOf(typeof(Ruleset)));

```

Comment:
you are permanently loading every dll in the folder. the overhead is not only huge, but you are wasting memory with a potentially second copy of EVERY dll in memory.